src/mame/machine/snes.cpp

```diff
@@ -230,14 +230,12 @@ TIMER_CALLBACK_MEMBER(snes_state::snes_hblank_tick)
 	/* draw a scanline */
 	if (m_ppu->current_vert() <= m_ppu->last_visible_line())
 	{
-		if (m_screen->vpos() > 0)
-		{
-			/* Do HDMA */
-			if (SNES_CPU_REG(HDMAEN))
-				hdma(cpu0space);
+		/* Do HDMA */
+		if (SNES_CPU_REG(HDMAEN))
+			hdma(cpu0space);
 
-			m_screen->update_partial((m_ppu->interlace() == 2) ? (m_ppu->current_vert() * m_ppu->interlace()) : m_ppu->current_vert() - 1);
-		}
+		if (m_screen->vpos() > 0)
+			m_screen->update_partial((m_ppu->interlace() == 2) ? (m_ppu->current_vert() * m_ppu->interlace()) : m_ppu->current_vert());
```

Comment:
Why did you get rid of the -1 on the non-interlaced path here?

Reply:
Hmm, should it update the previous scanline?

That generates a strange blue line in Vortex:
![0000](https://user-images.githubusercontent.com/3890995/65777785-f7a01500-e144-11e9-85ff-7ef5442e7f46.png)

Reply:
"This" being the behaviour with or without the -1 offset?  I was asking whether it was intentional because the guard for only doing it when the vertical position is non-zero shouldn't be needed if the -1 offset is removed.  It only makes sense to keep the check if you also keep the -1 offset.

Reply:
The line appears with the -1 offset. 
The guard still make sense because Snes starts the drawing in scanline 1 (nothing is
actually output for scanline 0).